hibernate-core/src/main/java/org/hibernate/query/criteria/internal/SelectionImplementor.java

```diff
@@ -15,4 +15,8 @@
  */
 public interface SelectionImplementor<X> extends TupleElementImplementor<X>, Selection<X>  {
 	public List<ValueHandlerFactory.ValueHandler> getValueHandlers();
+
+	default int getSelectionSpan() {
```

Comment:
I understand you wanted to be able to define a selection with a span of more than 1, but whose `isCompound` method would return `false`.

I wonder why exactly you don't want `MapEntryExpression` to be considered as compound? Does it break things, or do you fear that it will? After all, an entry is a key and a value, so it *is* compound...

Reply:
IIRC, that's the first thing I tried and it did break things.